qiskit_optimization/translators/gurobipy.py

```diff
@@ -169,14 +168,14 @@ def from_gurobipy(model: Model) -> QuadraticProgram:
     # keep track of names separately, since gurobipy allows to have None names.
     var_names = {}
     for x in model.getVars():
-        if x.vtype == gp.GRB.CONTINUOUS:
-            x_new = quadratic_program.continuous_var(x.lb, x.ub, x.VarName)
-        elif x.vtype == gp.GRB.BINARY:
+        if x.VType == gp.GRB.CONTINUOUS:
+            x_new = quadratic_program.continuous_var(x.LB, x.UB, x.VarName)
```

Comment:
Attributes with lower case still work with gurobipy 11.0. But, mypy does not pass due to the following reason.
gurobipy includes only a shared object until 10.0. On the other hand, gurobipy 11.0 includes a pyi file `gurobipy.pyi` as well as a shared object. The pyi file includes only the upper case attributes and mypy sees the file. So, we need to use upper case attributes or configure mypy to ignore the gurobipy pyi file.

Reply:
My comment was more around setup.py for the extra requires where the version is open and we say https://github.com/qiskit-community/qiskit-optimization/blob/7991f7f75bf5e12dd4b86fe0e7afce44c5e32fe1/setup.py#L79 In looking at the gurobi docs, at past versions, it seemed the uppercase LB, UB were there too as I noted above. If so then the extra requires is ok and the change is backwards compatible. Your explanation reads to me as though UB and LB only exist in 11.0 in which case, if that's the fix we want, shouldn't we limit the version there to >= 11.0

Reply:
Yes, gurobi docs at the past versions [e.g., 8.0](https://www.gurobi.com/documentation/8.0/refman/variable_attributes.html) include upper case attributes. So, we don't need add a version constraint. I intended to say that they started to provide a pyi file since 11.0 and it causes a strict type checking with mypy. But it does not change any runtime behavior.